compiler/rustc_typeck/src/errors.rs

```diff
@@ -324,3 +324,21 @@ pub struct SubstsOnOverriddenImpl {
     #[primary_span]
     pub span: Span,
 }
+
+#[derive(LintDiagnostic)]
+#[lint(typeck::unused_extern_crate)]
+pub struct UnusedExternCrate {
+    #[primary_span]
+    #[suggestion(applicability = "machine-applicable", code = "")]
+    pub span: Span,
+}
+
+#[derive(LintDiagnostic)]
+#[lint(typeck::extern_crate_not_idiomatic)]
+pub struct ExternCrateNotIdiomatic {
+    #[primary_span]
+    #[suggestion(applicability = "machine-applicable", code = "{suggestion_code}")]
```

Comment:
```suggestion
    #[suggestion_short(applicability = "machine-applicable", code = "{suggestion_code}")]
```

This should keep the output exactly the same as it was, so tests continue to pass.

Reply:
Thanks!